src/Database/Connection.php

```diff
@@ -727,6 +728,39 @@ protected function _newLogger(StatementInterface $statement)
     }
 
     /**
+     * Checks if the driver supports EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @return bool
+     */
+    public function canExplain($query)
+    {
+        $query = (string)$query;
+
+        return preg_match('/^\s*(SELECT|INSERT|UPDATE|DELETE)\s/i', $query)
+            && $this->_driver->explainSQL($query) !== false;
+    }
+
+    /**
+     * Executes an EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @param array $params list or associative array of params to be interpolated in $query as values
+     * @param array $types list or associative array of types to be used for casting values in query
+     * @return \Cake\Database\StatementInterface
+     */
+    public function explain($query, array $params = [], array $types = [])
```

Comment:
@lorenzo I know that `SqliteSchema::truncateTableSql()` is implemented in such way. But `truncate()` doesn't need to pass binding parameters, and doesn't need to fetch result set . If `explainSQL()` returns array, how do I handle it in DebugKit?
```php
$sqls = $driver->explainSQL($query);
foreach ($sqls as $sql => $flags) {
    if ($flags & SHOULD_PASS_ARGUMENTS) {
        $stmt = $connection->execute($sql, $params, $types); 
    } else {
        $stmt = $connection->execute($sql);
    }
    if ( $flags & SHOULD_FETCH) {
        $results = $stmt->fetchAll();
    }
}
```
Seems not elegant to me.